lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenPluginUtils.java

```diff
@@ -50,21 +50,30 @@ public static MarkupContent getMarkupDescription(Parameter parameter) {
 						+ LINE_BREAK + parameter.getDescription());
 	}
 	
-	// TODO: Handle the fact that MojoParameter's content (eg. DefaultValue) can be null
-	public static MarkupContent getMarkupDescription(MojoParameter parameter) {
-		return new MarkupContent("markdown",
-				"**required:** " + parameter.isRequired() + LINE_BREAK + "**Type:** " + parameter.getType() + LINE_BREAK
-						+ "Expression: " + parameter.getExpression() + LINE_BREAK + "Default Value: " + parameter.getDefaultValue()
-						+ LINE_BREAK + parameter.getDescription());
-	}
-	
-	// TODO: fix code duplication..
-	// TODO: add a note about using description and default valuefrom parent
-	public static MarkupContent getMarkupDescriptionUsingParent(MojoParameter parameter, MojoParameter parentParameter) {
-		return new MarkupContent("markdown",
-				"**required:** " + parameter.isRequired() + LINE_BREAK + "**Type:** " + parameter.getType() + LINE_BREAK
-						+ "Expression: " + parameter.getExpression() + LINE_BREAK + "Default Value: " + parentParameter.getDefaultValue()
-						+ LINE_BREAK + parentParameter.getDescription());
+	public static MarkupContent getMarkupDescription(MojoParameter parameter, MojoParameter parentParameter) {
+		final String fromParent = "**From parent configuration element:**" + LINE_BREAK;
+		String type = parameter.getType() != null ? parameter.getType() : "";
```

Comment:
Maybe `Optional` would be better here?

Reply:
Whatever works ;) I don't think it makes a big difference.

Reply:
Then this is fine for now ;)